src/EditorFeatures/CSharpTest/ValidateFormatString/ValidateFormatStringTests.cs

```diff
@@ -0,0 +1,625 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.ValidateFormatString;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ValidateFormatString
+{
+    public class ValidateFormatStringTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new ValidateFormatStringDiagnosticAnalyzer(), new EmptyCodeFixProvider());
+        
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task OnePlaceholder()
+        {
+            await TestMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""This {0} works[||]"", ""test""); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task TwoPlaceholders()
+        {
+            await TestMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""This {0} {1} works[||]"", ""test"", ""also""); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task ThreePlaceholders()
+        {
+            await TestMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""This {0} {1} works {2} [||]"", ""test"", ""also"", ""well""); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task FourPlaceholders()
+        {
+            await TestMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""This {1} is {2} [||]my {6} test "", ""teststring1"", ""teststring2"", ""teststring3"", ""teststring4"", ""teststring5"", ""teststring6"", ""teststring7"");
```

Comment:
Shouldn't this case show the diagnostic, since the placeholder is `6`, but there is not 7th value supplied?  It isn't about the absolute number of placeholders, but the highest index right?